bundles/org.eclipse.swt/Eclipse SWT Custom Widgets/common/org/eclipse/swt/custom/CTabFolderRenderer.java

```diff
@@ -362,8 +362,7 @@ protected Point computeSize (int part, int state, GC gc, int wHint, int hHint) {
 
 					if (parent.showClose || item.showClose) {
 						if ((state & SWT.SELECTED) != 0 || parent.showUnselectedClose) {
-							if (((state & SWT.SELECTED) != 0 && parent.showSelectedImage)
-									|| ((state & SWT.SELECTED) == 0 && parent.showUnselectedImage)) {
+							if (!applyLargeTextPadding(parent)) {
```

Comment:
this part deals with the regular padding (`INTERNAL_SPACING`). is correct to use the `applyLargeTextPadding()` condition here?

Reply:
Yes, this case distinction was introduced in #785. Before, the `INTERNAL_SPACING` was always added here. It is necessary in order to have some space between the text and the close icon. With large text padding, however, there already is sufficient space between the text and the close icon. Adding further spacing makes the text looks uncentered. In addition, the close icon already has some padding equal to `INTERNAL_SPACING`, which makes it look even more uncentered. Thus, we decided to not apply the additional spacing but subtract it in that case (see https://github.com/eclipse-platform/eclipse.platform.swt/pull/785/files#r1422065827).

Reply:
Ah, right. There are so many dimensions to this whitespace logic, it's hard to keep track of all of them.

Reply:
True. A refactoring could help to separate the different functionalities and improve comprehensibility, but that would be quite some effort and may be prone to errors...

Thanks for the review, @sratz!